ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlan-interfaces.j2

```diff
@@ -58,6 +58,12 @@ interface {{ vlan_interface.name }}
 {%     if vlan_interface.ip_verify_unicast_source_reachable_via is arista.avd.defined %}
    ip verify unicast source reachable-via {{ vlan_interface.ip_verify_unicast_source_reachable_via }}
 {%     endif %}
+{%     if vlan_interface.ip_dhcp_relay_all_subnets is arista.avd.defined(true) %}
+   ip dhcp relay all-subnets
+{%     endif %}
+{%     if vlan_interface.ipv6_dhcp_relay_all_subnets is arista.avd.defined(true) %}
+   ipv6 dhcp relay all-subnets
+{%     endif %}
```

Comment:
Please make sure to verify the order on EOS. Normally the ipv6 commands are not in the middle of ip ... commands.

Reply:
Same for the other config templates.

Reply:
For the global commands it is kind of mixed, I put it in a way I see it is closest to the final config. Or do you think it would make sense to have one shared template for IPv4 and IPv6?

```
ip dhcp relay information option
ip dhcp relay always-on
ip dhcp relay all-subnets default
ipv6 dhcp relay all-subnets default
ip dhcp relay qos dscp 43
no ipv6 dhcp relay qos dscp
ipv6 dhcp relay option link-layer address
ipv6 dhcp relay option remote-id format %m:%i
no ip dhcp relay access-group vrf MGMT
no ip dhcp relay access-group vrf HOSTING
no ip dhcp relay access-group
no ipv6 dhcp relay access-group vrf MGMT
no ipv6 dhcp relay access-group vrf HOSTING
no ipv6 dhcp relay access-group
```

For the interfaces it is also mixed, maybe because it is just one DhcpRelay agent for both IPv4 and IPv6:

```
interface Vlan10
   ...
   no bfd authentication mode
   ip dhcp relay all-subnets
   ipv6 dhcp relay all-subnets
   ip helper-address 2.2.2.1
   ipv6 dhcp relay destination 2002:b:1::10
   no ipv6 dhcp relay add vendor-option ccap-core
   no ipv6 dhcp relay install routes
   ip dhcp relay information option circuit-id Vlan10
   no dhcp relay ipv4 disabled
   no dhcp relay ipv6 disabled
   no dhcp server ipv4
   no dhcp server ipv6
   ...
```

Reply:
As long as you try to match EOS that is all I ask :) Thanks.